cli/src/semdep/find_lockfiles.py

```diff
@@ -110,3 +114,30 @@ def make_dependency_trie(
                 deps = list(parse_lockfile_str(lockfile.read_text(), lockfile))
                 dep_trie.insert(lockfile, deps, namespace)
     return dep_trie
+
+
+@lru_cache(maxsize=None)
```

Comment:
```suggestion
@lru_cache(maxsize=1000)
```

is this called once per lockfile pattern? we might wanna set this to like 100x the expected calls 1) in case a bug would result in a memory leak and 2) as sort of documentation about how much we expect this to be called.

Reply:
It's potentially called multiple times per lockfile pattern. If we have a yarn file in `A/` and we try to find the yarn file for `A/foo.js` and `A/B/bar.js`, then this function will get called on `A/` and `Yarn.lock` twice. So it should get called around `n * d` times, where `n` is the number of files with semgrep-core findings from reachability rules, and `d` is the average distance between a file and its lockfile in the directory tree.

Reply:
Ok I just got rid of this function and am not bothering with caching because checking if a file exists is very fast